src/services/factories/tokenList.ts

```diff
@@ -12,58 +12,103 @@ export function getTokensFactory(factoryParams: {
 
   const updatedIdsForNetwork = new Set<number>()
 
-  interface GetTokenIdResult {
-    value: number | null
+  interface UpdateTokenIdResult {
+    token: TokenDetails
     remove?: true
+    failed?: true
   }
 
-  async function getTokenId(networkId: number, tokenAddress: string): Promise<GetTokenIdResult> {
+  async function updateTokenId(networkId: number, token: TokenDetails): Promise<UpdateTokenIdResult> {
     try {
-      const hasToken = await exchangeApi.hasToken({ networkId, tokenAddress })
+      const hasToken = await exchangeApi.hasToken({ networkId, tokenAddress: token.address })
       if (!hasToken) {
-        return { value: null, remove: true }
+        return { token, remove: true }
       }
-      return { value: await exchangeApi.getTokenIdByAddress({ networkId, tokenAddress }) }
+      token.id = await exchangeApi.getTokenIdByAddress({ networkId, tokenAddress: token.address })
+      return { token }
     } catch (e) {
-      logDebug(`[network:${networkId}][address:${tokenAddress}] Failed to fetch id from contract`, e.message)
-      return { value: null }
+      logDebug(`[network:${networkId}][address:${token.address}] Failed to fetch id from contract`, e.message)
+      return { token, failed: true }
     }
   }
 
-  async function updateTokenIds(networkId: number): Promise<void> {
-    // Set as updated on start to prevent concurrent queries
-    updatedIdsForNetwork.add(networkId)
+  async function updateWithRetries(
+    networkId: number,
+    tokens: TokenDetails[],
+    maxRetries = 3,
+    retryDelay = 1000,
+    exponentialBackOff = true,
+  ): Promise<void> {
+    if (maxRetries <= 0) {
+      throw new Error(
+        `[network:${networkId}] Max retries exceeded trying to fetch token ids for tokens ${tokens.map(
+          t => t.address,
+        )}`,
+      )
+    }
 
-    const promises = tokenListApi.getTokens(networkId).map(token => getTokenId(networkId, token.address))
+    const promises = tokens.map(token => updateTokenId(networkId, token))
 
     const results = await Promise.all(promises)
```

Comment:
just wondering - what's the point of doing this if you're resolving right after? why not just
`const results = await Promise.all(tokens.map(token => updateTokenId(networkId, token)))`?

i could see it making sense if u were resolving much lower

Reply:
In a previous PR, @Velenir suggested to do it using `Promise.all` instead of what you are suggesting.
I honestly can't say what's better.